### PR TITLE
chore(*): update babylon version

### PR DESCRIPTION
**Pull request by Lazar955**
Updates babylond version 

[References issue ](https://github.com/babylonlabs-io/vigilante/issues/33)

### CODE REVIEW

File: e2etest/atomicslasher_e2e_test.go [hunk 1]
```diff
@@ -124,7 +121,9 @@ func TestAtomicSlasher(t *testing.T) {
 
 	require.Eventually(t, func() bool {
 		_, err := tm.BTCClient.GetRawTransaction(slashingTxHash2)
-		t.Logf("err of getting slashingTxHash of the BTC delegation affected by atomic slashing: %v", err)
+		if err != nil {
```
Comment (RafilxTenfen) [inline e2etest/atomicslasher_e2e_test.go:124]: :+1:

File: go.mod [hunk 2]
```diff
@@ -7,13 +7,14 @@ require (
 	cosmossdk.io/log v1.3.1
 	cosmossdk.io/math v1.3.0
 	github.com/avast/retry-go/v4 v4.6.0
-	github.com/babylonlabs-io/babylon v0.9.1
+	github.com/babylonlabs-io/babylon v0.9.3-0.20240904123958-b1e255a85e76
```
Comment (RafilxTenfen) [inline go.mod:10]: Is this version the latest main or latest released tag?

Comment (Lazar955) [reply go.mod:10]: I think the release tag, took this version as suggested by @KonradStaniec in [this PR](https://github.com/babylonlabs-io/btc-staker/pull/31#pullrequestreview-2280339986) to btc staker

**Review by RafilxTenfen — APPROVED** [review]
LGTM

**Pull request merged**